extensions/ql-vscode/src/cli-version.ts

```diff
@@ -6,12 +6,18 @@ import { Logger } from "./logging";
  * Get the version of a CodeQL CLI.
  */
 export async function getCodeQlCliVersion(codeQlPath: string, logger: Logger): Promise<semver.SemVer | undefined> {
-  const output: string = await runCodeQlCliCommand(
-    codeQlPath,
-    ["version"],
-    ["--format=terse"],
-    "Checking CodeQL version",
-    logger
-  );
-  return semver.parse(output.trim()) || undefined;
+  try {
+    const output: string = await runCodeQlCliCommand(
+      codeQlPath,
+      ["version"],
+      ["--format=terse"],
+      "Checking CodeQL version",
+      logger
+    );
+    return semver.parse(output.trim()) || undefined;
+  } catch (e) {
+    // failed to run the version command. This would happen if the cli version is _really_ old, or it is corrupted.
+    // in either we can't determine compatibility.
```

Comment:
Nit:

```suggestion
    // Failed to run the version command. This might happen if the cli version is _really_ old, or it is corrupted.
    // Either way, we can't determine compatibility.
```